runtime/nvqir/qpp/QppDMCircuitSimulator.cpp

```diff
@@ -69,11 +69,11 @@ class QppNoiseCircuitSimulator : public nvqir::QppCircuitSimulator<qpp::cmat> {
       return;
     }
 
-    state.conservativeResize(stateDimension, stateDimension);
-    for (std::size_t i = previousStateDimension; i < stateDimension; i++) {
-      state.col(i).setZero();
-      state.row(i).setZero();
-    }
+    // Additional qubit is added in the |0><0| state.
+    qpp::cmat zero_state = qpp::cmat::Zero(2, 2);
+    zero_state(0, 0) = 1.0;
+    state = qpp::kron(state, zero_state);
+    return;
```

Comment:
Nit: this `return` and the one on line 93 are not necessary and could be removed.